src/stats/probs/mod.rs

```diff
@@ -401,9 +402,10 @@ impl From<NotNan<f64>> for LogProb {
     }
 }
 
-impl From<LogProb> for NotNan<f64> {
-    fn from(p: LogProb) -> NotNan<f64> {
-        NotNan::new(*p).unwrap()
+impl TryFrom<LogProb> for NotNan<f64> {
```

Comment:
Sorry I was unclear but I would actually probably recommend leaving the (potentially-panicing) `From` implementation for now in order to avoid breaking compatibility, but mark it as deprecated.

Reply:
unfortunately you don't seem to be able to implement a deprecated `From` and a non-deprecated `TryFrom`, as the blanket implementation (https://github.com/rust-lang/rust/blob/3e525e3f6d9e85d54fa4c49b52df85aa0c990100/src/libcore/convert.rs#L580)  leads to the compiler thinking you've implemented `TryFrom` already and so it won't compile.

Reply:
I prefer the tryfrom here, let's make this a breaking change (feat!:), then.